### ISSUE TITLE
FAIL: datalad.support.tests.test_parallel.test_gracefull_death

### ISSUE THREAD

**Issue by mih** [datalad/datalad#5309]
Failed on 5th re-run of identical code.

```
Traceback (most recent call last):
2096  File "/Users/appveyor/venv3.8.6/lib/python3.8/site-packages/nose/case.py", line 198, in runTest
2097    self.test(*self.arg)
2098  File "/Users/appveyor/venv3.8.6/lib/python3.8/site-packages/datalad/tests/utils.py", line 263, in _wrap_skip_if
2099    return func(*args, **kwargs)
2100  File "/Users/appveyor/venv3.8.6/lib/python3.8/site-packages/datalad/support/tests/test_parallel.py", line 215, in test_gracefull_death
2101    assert_equal(sorted(inner()), [0, 1])
2102AssertionError: Lists differ: [1, 2] != [0, 1]
2103
2104First differing element 0:
21051
21060
2107
2108- [1, 2]
2109+ [0, 1]
```
- https://ci.appveyor.com/project/mih/datalad/build/job/alrk337e10quig3c#L2093
- https://ci.appveyor.com/project/mih/datalad/builds/37220881/job/hmnqn4ij7iqb5tgf#L2081
- https://ci.appveyor.com/project/mih/datalad/builds/37291931/job/1uqu1bw1sa7secdo#L2088

**Comment by yarikoptic** [comment]
FWIW: failed to reproduce on local OSX (yet to try to enter appveyor), or 1 core Linux VM (suspected that may be it is the reason) :-/  not sure if I will embark on appveyor debug route, so I will submit a PR to just announce it as a known/skip on OSX.  It isn't critical (just is testing expected behavior in case of being killed) so I think it is ok

**Comment by yarikoptic** [comment]
also could fail differently
```
FAILED ../datalad/support/tests/test_parallel.py::test_gracefull_death - assert 100 > 107

```

<details>
<summary>some grep to get some idea on how often happened and with what in the past month</summary> 

```shell
(git)smaug:/mnt/datasets/datalad/ci/logs/2022/08[master]git
$> git grep -l '^FAIL.*test_gracefull_dea' | xargs grep -B4 ': Assertion' 
03/pr/6915/964b1f4/travis-14378-failed/2.txt-        if msg is None:
03/pr/6915/964b1f4/travis-14378-failed/2.txt->           assert first > second
03/pr/6915/964b1f4/travis-14378-failed/2.txt-E           assert 100 > 121
03/pr/6915/964b1f4/travis-14378-failed/2.txt-
03/pr/6915/964b1f4/travis-14378-failed/2.txt:../datalad/tests/utils_pytest.py:119: AssertionError
--
03/pr/6915/964b1f4/travis-14378-failed/4.txt-        if msg is None:
03/pr/6915/964b1f4/travis-14378-failed/4.txt->           assert first in second
03/pr/6915/964b1f4/travis-14378-failed/4.txt-E           AssertionError: assert '.' in 'datalad 0+unknown\n'
03/pr/6915/964b1f4/travis-14378-failed/4.txt-
03/pr/6915/964b1f4/travis-14378-failed/4.txt:../../../../.local/lib/python3.10/site-packages/datalad/tests/utils_pytest.py:131: AssertionError
--
03/pr/6915/964b1f4/travis-14378-failed/4.txt-        if msg is None:
03/pr/6915/964b1f4/travis-14378-failed/4.txt->           assert first >= second
03/pr/6915/964b1f4/travis-14378-failed/4.txt-E           assert 0 >= 2
03/pr/6915/964b1f4/travis-14378-failed/4.txt-
03/pr/6915/964b1f4/travis-14378-failed/4.txt:../../../../.local/lib/python3.10/site-packages/datalad/tests/utils_pytest.py:125: AssertionError
--
03/pr/6915/964b1f4/travis-14378-failed/4.txt-E             +  '| We advise to re-install datalad or downstream projects might not operate '
03/pr/6915/964b1f4/travis-14378-failed/4.txt-E             +  'correctly. ',
03/pr/6915/964b1f4/travis-14378-failed/4.txt-E               ]
03/pr/6915/964b1f4/travis-14378-failed/4.txt-
03/pr/6915/964b1f4/travis-14378-failed/4.txt:../../../../.local/lib/python3.10/site-packages/datalad/tests/utils_pytest.py:107: AssertionError
--
23/pr/6958/fa7cee0/travis-14469-failed/9.txt-E             ?     ^
23/pr/6958/fa7cee0/travis-14469-failed/9.txt-E             + [0, 2]
23/pr/6958/fa7cee0/travis-14469-failed/9.txt-E             ?     ^
23/pr/6958/fa7cee0/travis-14469-failed/9.txt-
23/pr/6958/fa7cee0/travis-14469-failed/9.txt:../datalad/tests/utils_pytest.py:107: AssertionError
--
29/pr/6981/2551844/travis-14510-failed/3.txt-E             ?     ^
29/pr/6981/2551844/travis-14510-failed/3.txt-E             + [0, 2]
29/pr/6981/2551844/travis-14510-failed/3.txt-E             ?     ^
29/pr/6981/2551844/travis-14510-failed/3.txt-
29/pr/6981/2551844/travis-14510-failed/3.txt:../../../../.local/lib/python3.10/site-packages/datalad/tests/utils_pytest.py:107: AssertionError
--
30/push/maint/870c72a/travis-14519-failed/9.txt-E             Full diff:
30/push/maint/870c72a/travis-14519-failed/9.txt-E             - [0, 1]
30/push/maint/870c72a/travis-14519-failed/9.txt-E             + [1, 2]
30/push/maint/870c72a/travis-14519-failed/9.txt-
30/push/maint/870c72a/travis-14519-failed/9.txt:../datalad/tests/utils_pytest.py:107: AssertionError

```
</details>

it seems would be nice to make pytest to not shorten the error in the "summary" such as 
```
(git)smaug:/mnt/datasets/datalad/ci/logs/2022/08[master]git
$> git grep  '^FAIL.*test_gracefull_dea'                               
03/pr/6915/964b1f4/travis-14378-failed/2.txt:FAILED ../datalad/support/tests/test_parallel.py::test_gracefull_death - asse...
03/pr/6915/964b1f4/travis-14378-failed/4.txt:FAILED ../support/tests/test_parallel.py::test_gracefull_death - assert 0 >= 2
23/pr/6958/fa7cee0/travis-14469-failed/9.txt:FAILED ../datalad/support/tests/test_parallel.py::test_gracefull_death - asse...
29/pr/6981/2551844/travis-14510-failed/3.txt:FAILED ../support/tests/test_parallel.py::test_gracefull_death - assert [0, 2...
30/push/maint/870c72a/travis-14519-failed/9.txt:FAILED ../datalad/support/tests/test_parallel.py::test_gracefull_death - asse...
```

**Comment by yarikoptic** [comment]
was addressed by #7041
